packages/manager/modules/vps/src/upscale/configuration-price/configuration-price.controller.js

```diff
@@ -0,0 +1,20 @@
+import isString from 'lodash/isString';
+import PriceService from '../price/price.service';
+
+export default class {
+  $onInit() {
+    this.validateLanguage();
+    PriceService.validatePrice(this.price);
+
+    // the variables used in the view
+    this.v = {
+      price: PriceService.buildPriceToDisplay(this.price, this.language),
+    };
+  }
+
+  validateLanguage() {
+    if (!isString(this.language)) {
+      throw new TypeError('Attribute "language" should be of type "string"');
```

Comment:
if the language is of type string, why not use @ binding?

Reply:
Because it's a chore as long as you have to use bindings

```js
<configuration-price language="{{ $ctrl.theLanguageToUse}}"></configuration-price>
```
what a waste of time compared to
```js
<configuration-price language="$ctrl.theLanguageToUse"></configuration-price>
```

Also, you will never ever in your entire life

```js
<configuration-price language="fr_FR"></configuration-price>
```